lib/cdo/config.rb

```diff
@@ -0,0 +1,82 @@
+require 'singleton'
+require 'cdo/yaml'
+
+module Cdo
+  # Loads and combines structured application configuration settings.
+  class Config < OpenStruct
+    # Soft-freeze: Don't allow any config items to be created/modified,
+    # but allow stubbing for unit tests.
+    def freeze
+      @table.each_key(&method(:new_ostruct_member!))
+      @frozen = true
+    end
+
+    def method_missing(key, *args)
+      msg = "Undefined #{self.class} reference: #{key}"
+      raise ArgumentError, msg, caller(1) if @frozen
+      # puts msg, caller(1) if @frozen
```

Comment:
Remove? (And in `modifiable?` also)

Reply:
oh yeah, that was leftover from manual testing, I will clean this up.